src/Stryker.Core/Stryker.Core/Initialisation/InitialisationProcess.cs

```diff
@@ -79,26 +82,59 @@ public MutationTestInput Initialize(StrykerOptions options)
                     options.MsBuildPath);
             }
 
-            InitializeDashboardProjectInformation(options, projectInfo);
+            InitializeDashboardProjectInformation(options, _projectInfo);
 
             if (_testRunner == null)
             {
-                _testRunner = new VsTestRunnerPool(options, projectInfo);
+                _testRunner = new VsTestRunnerPool(options, _projectInfo);
             }
 
             var input = new MutationTestInput
             {
-                ProjectInfo = projectInfo,
-                AssemblyReferences = _assemblyReferenceResolver.LoadProjectReferences(projectInfo.ProjectUnderTestAnalyzerResult.References).ToList(),
+                ProjectInfo = _projectInfo,
+                AssemblyReferences = _assemblyReferenceResolver.LoadProjectReferences(_projectInfo.ProjectUnderTestAnalyzerResult.References).ToList(),
                 TestRunner = _testRunner,
             };
 
             return input;
         }
 
-        public InitialTestRun InitialTest(StrykerOptions options) =>
+        public InitialTestRun InitialTest(StrykerOptions options)
+        {
             // initial test
-            _initialTestProcess.InitialTest(options, _testRunner);
+            var result = _initialTestProcess.InitialTest(options, _testRunner);
+
+            if (_testRunner.DiscoverTests().Count != 0)
```

Comment:
We should rename this at some point because testrunner.discoverTests never actually runs test discovery anymore.